### PR TITLE
[DOCS] Adds limitation on deprecated painless scripts and edits limitation regarding spaces in Transform docs

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR applies the following changes on the transform limitations page:
* adds a limitation about deprecation messages for transforms that use Painless with deprecated syntax,
* edits a limitation regarding Kibana spaces.

### Preview

[Deprecation warnings for Painless scripts](https://elasticsearch_86397.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/transform-limitations.html#transform-painless-warning-limitation)
[Spaces limitation](https://elasticsearch_86397.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/transform-limitations.html#transform-space-limitations)

### CODE REVIEW

File: docs/reference/transform/limitations.asciidoc [hunk 1]
```diff
@@ -277,8 +288,10 @@ documents if your {transform} contains a `group_by` based on `date_histogram`.
 
 {kibana-ref}/xpack-spaces.html[Spaces] enable you to organize your source and 
 destination indices and other saved objects in {kib} and to see only the objects 
-that belong to your space. However, this limited scope does not apply to 
-{transforms}; they are visible in all spaces.
+that belong to your space. However, a {transform} is a long running task which 
+is managed on cluster level and therefore not limited in scope to certain 
+spaces. Space awareness can be implemented by the creation of the {data-source}
```
Comment (lcawl) [inline docs/reference/transform/limitations.asciidoc:293]: > Space awareness can be implemented by the creation of the {data-source} 

Can we provide a link or more info? I didn't think space-level privileges were associated with indices or data views. However, if we're just referring to index-level privileges, we can link to https://www.elastic.co/guide/en/kibana/master/kibana-role-management.html

**Review by przemekwitek — APPROVED** [review]
LGTM

File: docs/reference/transform/limitations.asciidoc (hunk 1 above)
Comment (szabosteve) [reply docs/reference/transform/limitations.asciidoc]: @lcawl It is possible to assign Spaces to data views or indices as the screenshot below shows.
Screenshot 2022-05-04 at 11 12 24
I provided some more info via 
[057207f](https://github.com/elastic/elasticsearch/pull/86397/commits/057207f625d451fa3a1066f395acf883b211cf3d).
Please let me know what you think!

Comment (lcawl) [reply docs/reference/transform/limitations.asciidoc]: TIL!  Thanks for the clarification!

**Review by lcawl — APPROVED** [review]
LGTM

**Pull request merged**